File: examples/35-prelude-helpers/main.go
package main

import (
	"fmt"
	"os"
	"path/filepath"

	log "github.com/djinn-soul/gopptx/pkg/stdlog"

	"github.com/djinn-soul/gopptx/pkg/pptx"
	"github.com/djinn-soul/gopptx/pkg/pptx/styling"
)

const (
	outputDir  = "examples/output"
	outputFile = "35_prelude_helpers.pptx"
)

func main() {
	if err := run(); err != nil {
		fmt.Fprintf(os.Stderr, "error: %v\n", err)
		os.Exit(1)
	}
}

func run() error {
	if err := os.MkdirAll(outputDir, 0o750); err != nil {
		return fmt.Errorf("create output dir: %w", err)
	}

	builder := pptx.NewPresentationBuilder("Prelude Helpers Demo")

	// Set widescreen 16:9 slide size via the custom SlideSize helper.
	// WithSlideSize accepts a SlideSize value; SlideSize16x9() returns the
	// standard widescreen preset (approximately 16in × 9in in EMU).
	builder.WithSlideSize(pptx.SlideSize16x9())

	// Apply a theme using the fluent builder.
	builder.WithTheme(styling.ThemeModern)

	// Slide 1: welcome / overview
	builder.AddSlide(
		pptx.NewSlide("Welcome").
			AddBullet("Built with PresentationBuilder").
			AddBullet("Fluent API for creating presentations").
			AddBullet("Chain methods for concise setup"),
	)

	// Slide 2: numbered list of features
	builder.AddSlide(
		pptx.NewSlide("Features").
			AddNumbered("Builder pattern").
			AddNumbered("Theme support").
			AddNumbered("Slide size control").
			AddNumbered("Metadata helpers"),
	)

	// Slide 3: slide size awareness – show the EMU values of a 16:9 slide
	w16 := styling.Inches(16)
	h9 := styling.Inches(9)
	builder.AddSlide(
		pptx.NewSlide("Slide Size Awareness").
			AddBullet(fmt.Sprintf("16 inches wide = %d EMU", w16.Emu())).
			AddBullet(fmt.Sprintf(" 9 inches tall = %d EMU", h9.Emu())).
			AddBullet("styling.Inches() converts from human-readable to EMU").
			AddBullet("SlideSize16x9() returns the standard preset"),
	)

	// Slide 4: metadata and slide numbers
	builder.WithSlideNumbers(true)
	builder.AddSlide(
		pptx.NewSlide("Builder Options").
			AddBullet("WithSlideNumbers(true) – page numbers on every slide").
			AddBullet("WithFooter(text) – footer across all slides").
			AddBullet("WithTheme(theme) – apply a color+font palette").
			AddBullet("WriteToFile(path) – build and persist in one call"),
	)

	outputPath := filepath.Join(outputDir, outputFile)
	if err := builder.WriteToFile(outputPath); err != nil {
		return fmt.Errorf("write: %w", err)
	}

	log.Printf("Generated %s\n", outputPath)
	return nil
}

File: examples/35-prelude-helpers/main.py
"""Demonstrate PresentationBuilder helpers: themes, slide size, slide numbers.

This example demonstrates:
- Setting 16:9 slide size
- Applying a built-in theme preset
- Using slide numbers and footers
- Builder pattern for concise presentation setup
"""

from __future__ import annotations

from pathlib import Path

from gopptx import Presentation
from gopptx.constants import SIZE_16X9_HEIGHT, SIZE_16X9_WIDTH
from gopptx.presentation.theme import get_theme


def main() -> None:
    """Create presentation demonstrating prelude helper utilities."""
    output_dir = Path("examples/output")
    output_dir.mkdir(exist_ok=True)

    with Presentation.new("Prelude Helpers Demo") as prs:
        # Set widescreen 16:9 slide size
        prs.set_slide_size(SIZE_16X9_WIDTH, SIZE_16X9_HEIGHT)

        # Apply built-in theme preset
        prs.apply_theme(get_theme("ocean"))

        # Slide 1: welcome / overview
        prs.add_bullet_slide(
            "Welcome",
            [
                "Built with PresentationBuilder",
                "Fluent API for creating presentations",
                "Chain methods for concise setup",
            ],
        )

        # Slide 2: numbered features list
        prs.add_bullet_slide(
            "Features",
            [
                "1. Builder pattern",
                "2. Theme support",
                "3. Slide size control",
                "4. Metadata helpers",
            ],
        )

        # Slide 3: slide size awareness — EMU values
        emu_16 = int(SIZE_16X9_WIDTH)
        emu_9 = int(SIZE_16X9_HEIGHT)
        prs.add_bullet_slide(
            "Slide Size Awareness",
            [
                f"16:9 width = {emu_16} EMU",
                f"16:9 height = {emu_9} EMU",
                "Inches() converts from human-readable to EMU",
                "SIZE_16X9_WIDTH/HEIGHT are standard presets",
            ],
        )

        # Slide 4: builder options
        prs.add_bullet_slide(
            "Builder Options",
            [
                "set_slide_size(w, h) — widescreen or custom dimensions",
                "apply_theme(theme) — apply a color+font palette",
                "set_core_properties({...}) — core properties",
                "save(path) — build and persist in one call",
            ],
        )

        output_path = output_dir / "35_prelude_helpers.pptx"
        prs.save(str(output_path))
        print(f"Saved: {output_path}")

    print("\n=== SUMMARY ===")
    print("4 slides: Welcome, Features, Slide Size, Builder Options")
    print(f"16:9 dimensions: {SIZE_16X9_WIDTH} x {SIZE_16X9_HEIGHT} EMU")


if __name__ == "__main__":
    main()
